client/internal/engine.go

```diff
@@ -770,6 +771,12 @@ func (e *Engine) receiveSignalEvents() {
 				return fmt.Errorf("wrongly addressed message %s", msg.Key)
 			}
 
+			if msg.GetBody().GetPort() != 0 {
+				conf := conn.GetConf()
+				conf.ProxyConfig.WgPort = int(msg.GetBody().GetPort())
+				conn.UpdateConf(conf)
+			}
+
```

Comment:
I would rather keep this logic inside the conn.
It is not a concern of the Engine to decide.